packages/fuels-core/src/code_gen/custom_types_gen.rs

```diff
@@ -731,4 +1185,179 @@ mod tests {
         assert_eq!(actual, expected);
         Ok(())
     }
+
+    #[test]
+    fn test_expand_struct_new_abi() -> Result<(), Error> {
+        let s = r#"
+        {
+            "types": [
+              {
+                "typeId": 6,
+                "type": "u64",
+                "components": null,
+                "typeParameters": null
+              },
+              {
+                "typeId": 8,
+                "type": "b256",
+                "components": null,
+                "typeParameters": null
+              },
+              {
+                "typeId": 6,
+                "type": "u64",
+                "components": null,
+                "typeParameters": null
+              },
```

Comment:
Although it would be much more readable, it's not the reality of the compiler at the moment. Type duplication will be present in the JSON ABI, and we should be able to handle that. Also, it won't have an apparent order for reasons internal to the compiler. All of this might (should) change in the future, but we should be able to handle it without issues for now. This example is an actual compiler output of one of our test projects!